### PR TITLE
Additional Remarks - Additional changes in behavior - 781a

### PR DESCRIPTION
**Pull request by imeecuison**
## Description
Option to enter additional remarks in behavior - particularly social behavior changes.
These questions are asked after Additional Event screen, physical, mental, and work behavior changes screens.
```bah-781a-additionalRemarks-additional```
http://localhost:3001/disability-benefits/apply/form-526-all-claims/new-disabilities/ptsd-781a-additional-changes

## Testing done
tests/pages/additionalBehaviorChanges.unit.spec.jsx

## Screenshots
screen shot 2018-11-16 at 10 14 08 am

## Acceptance criteria
- [ ] The PTSD type classification is correctly populated on page header based on the user's selection at the beginning of the workflow
- [ ] User is able to add additional information if needed, per the Content Document
**NOTED: No error messages will appear because this form has no required fields.**

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by devin-mccurdy — CHANGES_REQUESTED** [review]
Looks good, but can you add code coverage for `isAnsweringPtsdForm` in utils?

**Comment by imeecuison** [comment]
@dmccurdybah : I added tests for isAnsweringPtsdForm. Thanks!

**Review by devin-mccurdy — APPROVED** [review]

**Comment by imeecuison** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group Could I get a code review on this at your earliest convenience.

File: src/applications/disability-benefits/all-claims/pages/additionalBehaviorChanges.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      rows: 5,
+      maxLength: 32000,
+    },
+  },
+};
+
+export const schema = {
+  type: 'object',
+  properties: {
+    additionalChanges: {
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/pages/additionalBehaviorChanges.js]: I think this field needs to be added to `all-claims/config/schema.js`?

**Comment by jbalboni** [comment]
@imeecuison I think you need to merge in the latest from master to get the build to pass.

**Review by jbalboni — COMMENTED** [review]
One question, but mostly looks fine.

File: src/applications/disability-benefits/all-claims/pages/additionalBehaviorChanges.js (hunk 1 above)
Comment (imeecuison) [reply src/applications/disability-benefits/all-claims/pages/additionalBehaviorChanges.js]: @jbalboni: I've addd to all-claims/config/shema.js
Thank you!

**Review by jbalboni — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx [hunk 2]
```diff
@@ -482,6 +483,20 @@ describe('526 helpers', () => {
     });
   });
 });
+<<<<<<< HEAD
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx]: There are merge conflict symbols in this file.

File: src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx [hunk 3]
```diff
@@ -482,6 +483,20 @@ describe('526 helpers', () => {
...
+    const formData = {
+      'view:uploadPtsdChoice': 'answerQuestions',
+    };
+    expect(isAnsweringPtsdForm(formData)).to.be.true;
+  });
+
+  it('should return false if user has chosen to not answer questions', () => {
+    const formData = {};
+    expect(needsToEnter781({ formData })).to.be.false;
+=======
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx]: Merge conflict symbol

File: src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx [hunk 4]
```diff
@@ -542,5 +557,6 @@ describe('isAnswering781aQuestions', () => {
       'view:enterAdditionalSecondaryEvents0': false,
     };
     expect(isAnswering781aQuestions(1)(formData)).to.be.false;
+>>>>>>> e8f9324053a40baab4e31d8df6c81a5d9f6956dc
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx]: merge conflict symbol

**Review by rianfowler — CHANGES_REQUESTED** [review]
LMK if you run into Jenkins failures after repairing these files.

File: src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx (hunk 4 above)
Comment (imeecuison) [reply src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx]: @rianfowler : Fixed those merge conflicts. Thank you!

**Comment by imeecuison** [comment]
@rianfowler : Can you please take a look at this when you get a chance? Thank you!

**Review by rianfowler — APPROVED** [review]

**Review by rianfowler — CHANGES_REQUESTED** [review]

File: src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx [hunk 5]
```diff
@@ -23,6 +23,7 @@ import {
   isUploading781Form,
   isUploading781aForm,
   transformRelatedDisabilities,
+  isAnsweringPtsdForm,
```
[the remark this answers is not in the document]
Comment (imeecuison) [reply src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx]: removed! thank you!

[the remark this answers is not in the document]
Comment (imeecuison) [reply src/applications/disability-benefits/all-claims/tests/utils.unit.spec.jsx]: @rianfowler : please review again when you get a chance. Thank you!

**Review by rianfowler — APPROVED** [review]

**Comment by imeecuison** [comment]
@aaronjlech : Can you look at this really quick to confirm use of depends changes?

**Review by aaronjlech — CHANGES_REQUESTED** [review]

**Review by aaronjlech — CHANGES_REQUESTED** [review]
@imeecuison Can we update these two functions in the utils, they are out of date with the changes made to upload Properties.

**OLD**
```
 
 export const showPtsdCombatConclusion = form =>
   form['view:uploadPtsdChoice'] === 'answerQuestions' &&
   (_.get('view:selectablePtsdTypes.view:combatPtsdType', form, false) ||
     _.get('view:selectablePtsdTypes.view:noncombatPtsdType', form, false));
 
 export const showPtsdAssaultConclusion = form =>
   form['view:uploadPtsdChoice'] === 'answerQuestions' &&
   (_.get('view:selectablePtsdTypes.view:mstPtsdType', form, false) ||
     _.get('view:selectablePtsdTypes.view:assaultPtsdType', form, false));
```

**NEW**
```
 
 export const showPtsdCombatConclusion = form =>
   form['view:upload781Choice'] === 'answerQuestions' &&
   (_.get('view:selectablePtsdTypes.view:combatPtsdType', form, false) ||
     _.get('view:selectablePtsdTypes.view:noncombatPtsdType', form, false));
 
 export const showPtsdAssaultConclusion = form =>
   form['view:upload781aChoice'] === 'answerQuestions' &&
   (_.get('view:selectablePtsdTypes.view:mstPtsdType', form, false) ||
     _.get('view:selectablePtsdTypes.view:assaultPtsdType', form, false));
```

**Review by aaronjlech — APPROVED** [review]
LGTM!

**Pull request merged**